HiAnalysis/HiOnia/plugins/HiOniaAnalyzer.cc

```diff
@@ -209,33 +263,6 @@ class HiOniaAnalyzer : public edm::EDAnalyzer {
   int  Reco_QQ_NtrkDeltaR04[Max_QQ_size];
   int  Reco_QQ_NtrkDeltaR05[Max_QQ_size];
 
-  bool Reco_QQ_mupl_isGoodMuon[Max_QQ_size];      // Vector of isGoodMuon(TMOneStationTight) for plus muon
-  bool Reco_QQ_mumi_isGoodMuon[Max_QQ_size];      // Vector of isGoodMuon(TMOneStationTight) for minus muon
-  bool Reco_QQ_mupl_highPurity[Max_QQ_size];      // Vector of high purity flag for plus muon
-  bool Reco_QQ_mumi_highPurity[Max_QQ_size];      // Vector of high purity flag for minus muon
-  bool Reco_QQ_mupl_TrkMuArb[Max_QQ_size];      // Vector of TrackerMuonArbitrated for plus muon
-  bool Reco_QQ_mumi_TrkMuArb[Max_QQ_size];      // Vector of TrackerMuonArbitrated for minus muon
-  bool Reco_QQ_mupl_TMOneStaTight[Max_QQ_size]; // Vector of TMOneStationTight for plus muon
-  bool Reco_QQ_mumi_TMOneStaTight[Max_QQ_size]; // Vector of TMOneStationTight for minus muon
-
-  int Reco_QQ_mupl_SelectionType[Max_QQ_size];  
-  int Reco_QQ_mumi_SelectionType[Max_QQ_size];  
-  int Reco_QQ_mupl_nPixValHits[Max_QQ_size];  // Number of valid pixel hits in plus sta muons
-  int Reco_QQ_mumi_nPixValHits[Max_QQ_size];  // Number of valid pixel hits in minus sta muons
-  int Reco_QQ_mupl_nMuValHits[Max_QQ_size];  // Number of valid muon hits in plus sta muons
-  int Reco_QQ_mumi_nMuValHits[Max_QQ_size];  // Number of valid muon hits in minus sta muons
-  int Reco_QQ_mupl_nTrkHits[Max_QQ_size];  // track hits plus global muons
-  int Reco_QQ_mumi_nTrkHits[Max_QQ_size];  // track hits minus global muons
-  int Reco_QQ_mupl_nPixWMea[Max_QQ_size];  // pixel layers with measurement for plus inner track muons
-  int Reco_QQ_mumi_nPixWMea[Max_QQ_size];  // pixel layers with measurement for minus inner track muons
-  int Reco_QQ_mupl_nTrkWMea[Max_QQ_size];  // track layers with measurement for plus inner track muons
-  int Reco_QQ_mumi_nTrkWMea[Max_QQ_size];  // track layers with measurement for minus inner track muons
-  int Reco_QQ_mupl_StationsMatched[Max_QQ_size];  // number of stations matched for plus inner track muons
-  int Reco_QQ_mumi_StationsMatched[Max_QQ_size];  // number of stations matched for minus inner track muons
-  float Reco_QQ_mupl_normChi2_inner[Max_QQ_size];  // chi2/ndof for plus inner track muons
-  float Reco_QQ_mumi_normChi2_inner[Max_QQ_size];  // chi2/ndof for minus inner track muons
-  float Reco_QQ_mupl_normChi2_global[Max_QQ_size];  // chi2/ndof for plus global muons
-  float Reco_QQ_mumi_normChi2_global[Max_QQ_size];  // chi2/ndof for minus global muons
   float Reco_QQ_mupl_dxy[Max_QQ_size];  // dxy for plus inner track muons
   float Reco_QQ_mumi_dxy[Max_QQ_size];  // dxy for minus inner track muons
   float Reco_QQ_mupl_dxyErr[Max_QQ_size];  // dxy error for plus inner track muons
```

Comment:
If you check the moment I fill the tree, I renamed these variables "Reco_QQ_muXX_dXX_muonlessVtx", which is more explicit.

So the reason is that, when we use the muonless primary vertex for a given dimuon, the PV will change, so the dxy and dz will change as well. Reco_QQ_muXX_dXX_muonlessVtx gives you the dXX for this newly found vertex.
If you use Reco_mu_dXX[Reco_QQ_muXX_idx], it will give you the dz or dxy for the OLD primary vertex, without refitting.